src/bindgen/ir/enumeration.rs

```diff
@@ -321,6 +322,22 @@ impl Enum {
         repr.style != ReprStyle::C
     }
 
+    fn typedef_inline_enum_macro(&self, config: &Config) -> Option<&str> {
+        let is_c = config.language == Language::C;
```

Comment:
Why don't we want to generate this for C++?

Reply:
Since this is primarily targeted at ObjC and Swift and people using the (Core)Foundation headers I figure people aren't using C++ output for that. I could be wrong, but I don't see it being useful in that case. All that's at stake is making people remove this from the configuration when switching language, I guess.

I added that line originally to avoid code path in Python mode when writing the "ctypedef" line.

Reply:
The unmentioned context is that

1. technically yes, NS_ENUM will work fine in Apple toolchains in C++ mode, I believe it will produce unscoped enums with fixed types just like cpp_compat currently does; but
2. targeting C++ presently gives you enum classes, which are better in literally every way than NS_ENUM.

Basically we can probably afford to be smarter than our users here.

Final thought is along similar lines: The cpp_compat option clearly considers that the C output is meant to be used with `[enum] prefix_with_name = true`, as is customary for C. Hence the use of unscoped fixed type enums (`enum Name : type` instead of scoped `enum class Name : type`) in the ifdef. Should that option, which currently defaults to false to match C++ scoped enums, default to true instead for C?

Reply:
Ah, one more thing, Swift cannot import C++ and all C++ interop currently has to be done via writing or generating C bindings for the C++ code. There is a [document ('manifesto') with some design considerations][manifesto] for a potential Swift/C++ interop, but if Swift ever gets to import C++ directly, it will obviously address fixed-type enums and enum classes. So NS_ENUM for C++ output will never be necessary.

[manifesto]: https://github.com/apple/swift/blob/main/docs/CppInteroperabilityManifesto.md

Reply:
Ok, sounds good, thanks.